src/ngraph/ngraph_graph.h

```diff
@@ -219,6 +219,7 @@ class Graph : public Node {
       ngraph_forward[i] = nullptr;
       ngraph_backward[i] = nullptr;
     }
+    fprop_cache = ngraph::FpropCache();
```

Comment:
Is this the last time we'll actually want a populated `fprop_cache` for this `ngraph_bridge::Graph` object?

If so, I wonder if it would be a little more intuitive to have `fprop_cache` be a `shared_ptr<ngraph::FpropCache>`, and we'd replace this statement with setting the pointer to null?